lib/cocoapods-core/specification/dsl.rb

```diff
@@ -76,8 +76,10 @@ module DSL
       #   @param  [String] name
       #           the name of the pod.
       #
-      root_attribute :name,
-                     :required => true
+      attribute :name,
```

Comment:
Agree on that, but if it's defined as `root_attribute`, it will be never validated for subspecs and the behavior, when the value is set, is undefined. It would be an exception among other root attributes, because it may occur on subspec level. I think being to overwrite it in the subspec block in the DSL is the lesser of the two evils.